docs: add information about the build system (#1299)
coreos/ignition: docs/development.md

Create a new [release checklist](https://github.com/coreos/ignition/issues/new?labels=kind/release&template=release-checklist.md) and follow the steps there.

## The build process

> Note that the `build` script included in this repository is a convenience script only and not used for the actual release binaries. Those are built using a `ignition.spec` maintained in [Fedora rpms/ignition](https://src.fedoraproject.org/rpms/ignition).

Reviewer: The published ignition-validate container isn't built from the specfile; the Dockerfile uses `./build_for_container`.

nit: "an `ignition.spec`"

Author: Should be fixed now

Create a new [release checklist](https://github.com/coreos/ignition/issues/new?labels=kind/release&template=release-checklist.md) and follow the steps there.

## The build process

> Note that the `build` script included in this repository is a convenience script only and not used for the actual release binaries. Those are built using an `ignition.spec` maintained in [Fedora rpms/ignition](https://src.fedoraproject.org/rpms/ignition) and the `build_for_container` script in case of the ignition-validate container, respectively.

Reviewer: Could we make it clearer that the rest of this section covers `ignition.spec` and not `build_for_container`?  Maybe something like:

```suggestion
Note that the `build` script included in this repository is a convenience script only and not used for the actual release binaries. Those are built using an `ignition.spec` maintained in [Fedora rpms/ignition](https://src.fedoraproject.org/rpms/ignition). (The `ignition-validate` [container](https://quay.io/repository/coreos/ignition-validate) is built by the `build_for_container` script, which is not further described here.)
```

Author: Applied your changes :)
